src/main/java/com/oltpbenchmark/api/collectors/monitoring/PostgreSQLMonitor.java

```diff
@@ -0,0 +1,151 @@
+import com.oltpbenchmark.util.MonitorInfo;
+import com.oltpbenchmark.util.MonitoringUtil;
+
+/**
+ * Implementation of a monitor specific to PostgreSQL. Uses the
+ * 'pg_stat_statements' add-on to extract relevant query and system information.
+ */
+public class PostgreSQLMonitor extends DatabaseMonitor {
+
+    private final String PG_STAT_STATEMENTS = """
+        SELECT query AS query_text, calls as execution_count, rows, 
+        total_exec_time, min_exec_time, max_exec_time, 
+        shared_blks_read, shared_blks_written, local_blks_read, 
+        local_blks_written, temp_blks_read, temp_blks_written 
+        FROM pg_stat_statements;
+        """;
+    private final String CLEAN_CACHE = "SELECT pg_stat_statements_reset();";
+    private final List<String> singleQueryProperties;
+    private final List<String> repeatedQueryProperties;
+
+    private final Set<String> stored_queries;
+
+    public PostgreSQLMonitor(MonitorInfo monitorInfo, BenchmarkState testState,
+            List<? extends Worker<? extends BenchmarkModule>> workers, WorkloadConfiguration conf) {
+        super(monitorInfo, testState, workers, conf);
+
+        this.stored_queries = new HashSet<String>();
+
+        this.singleQueryProperties = new ArrayList<String>() {
+            {
+                add("query_text");
+            }
+        };
+
+        this.repeatedQueryProperties = new ArrayList<String>() {
+            {
+                add("execution_count");
+                add("min_exec_time");
+                add("max_exec_time");
+                add("total_exec_time");
+                add("rows");
+                add("shared_blks_read");
+                add("shared_blks_written");
+                add("local_blks_read");
+                add("local_blks_written");
+                add("temp_blks_read");
+                add("temp_blks_written");
+            }
+        };
+    }
+
+    /**
+     * Extract query events (single and repeated) using the extraction query
+     * and properties defined above.
+     */
+    private void extractQueryMetrics(Instant instant) {
+        ImmutableSingleQueryEvent.Builder singleQueryEventBuilder = ImmutableSingleQueryEvent.builder();
+        ImmutableRepeatedQueryEvent.Builder repeatedQueryEventBuilder = ImmutableRepeatedQueryEvent.builder();
+
+        try (PreparedStatement stmt = conn.prepareStatement(PG_STAT_STATEMENTS)) {
+            ResultSet rs = stmt.executeQuery();
+            while (rs.next()) {
+                // Only store those queries that have monitoring enabled via a
+                // comment in the SQL Server dialect XML.
+                String query_text = rs.getString("query_text");
+                if (!query_text.contains(MonitoringUtil.getMonitoringPrefix())) {
+                    continue;
+                }
+                // Get identifier from commment in query text.
+                String[] split = query_text.split(Pattern.quote(MonitoringUtil.getMonitoringPrefix()));
```

Comment:
Since you're already using regexes, could you not use a regex capture to pull the string name out and drop the splitting and suffix tracking?

I think the first prefix and the simple `if (query_text.contains(MonitoringUtil.getMonitoringPrefix()))` makes sense as an efficient initial filter if you expect many items not to match, but then after that you could probably do something like this for the queryId extraction:

```java
class MonitoringUtil {
    public static final Pattern monitoringCommentPattern = Pattern.compile("/[*] monitoring-(\S+) */");

    // or even with named captures
    public static final Pattern MonitoringCommentPattern = Pattern.compile("/[*] monitoring-(?<queryId>\S+) */");
    // the exact regex may require tweaking if you don't have the surrounding whitespace in the comment text for instance
```

Now, when you go to match the query text you could do something like this:
```java
Matcher m = MonitoringUtil.MonitoringCommentPattern.matcher(query_text)
if (m.find()) {
    queryId = m.group(1);
    // or if using named capture group
    queryId = m.group("queryId");  // I think this potentially has slightly higher overhead, but I find it more readable.
}
```